src/signature_util.cc

```diff
@@ -101,7 +98,9 @@ bool SignatureUtil::verifySignature(std::string_view bytecode, std::string &mess
   uint8_t hash[SHA512_DIGEST_LENGTH];
   SHA512_Final(hash, &ctx);
 
-  static const auto ed25519_pubkey = hex2pubkey<32>(PROXY_WASM_VERIFY_WITH_ED25519_PUBKEY);
+  char pubkey_char[65];
+  strcpy(pubkey_char, pubkey.c_str());
+  static const auto ed25519_pubkey = hex2pubkey<32>(pubkey_char);
```

Comment:
This is `static` variable, so the first key used would persist forever and it would be used for all verifications, leading to false negatives.